SecurityPkg/VariableAuthenticated/SecureBootConfigDxe/SecureBootConfigImpl.c

```diff
@@ -2652,8 +2665,37 @@ UpdateDeletePage (
       //
       // Display GUID and help
       //
-      GuidToString (&Cert->SignatureOwner, GuidStr, 100);
-      GuidID = HiiSetString (PrivateData->HiiHandle, 0, GuidStr, NULL);
+
+      CertificateInfoReadSuccess = X509GetSubjectName (
+        (UINT8*)Cert->SignatureData,
+        (UINTN)CertList->SignatureSize,
+        (UINT8*)CertificateInfoStr8,
+        &CertificateInfoStrSize
+      );
+      DEBUG(("CertinfoSize: %d", CertificateInfoStrSize));
+
+      // if (!CertificateInfoReadSuccess) {
+      // CertificateInfoReadSuccess = X509GetCommonName (
+      //   (UINT8*)Cert->SignatureData,
+      //   (UINTN)CertList->SignatureSize,
+      //   CertificateInfoStr8,
+      //   &CertificateInfoStrSize
+      // );
+      // // }
+
+      CertificateInfoReadSuccess = CertificateInfoReadSuccess && (AsciiStrLen(CertificateInfoStr8) >= 2);
+
+      if(CertificateInfoReadSuccess) {
+        CertificateInfoStr8[CertificateInfoStrSizeHalf] = '\0';
+        DEBUG(("cert8: %s", CertificateInfoStr8));
+        AsciiStrToUnicodeStrS(CertificateInfoStr8, CertificateInfoStr, CertificateInfoStrSize/sizeof(CHAR16));
+      } else {
+      StrCpyS(CertificateInfoStr, CertificateInfoStrSize/sizeof(CHAR16), UNKNOWN_CERT_PLACEHOLDER);
+      }

```

Comment:
Maybe instead of copying this static UNKNOWN_CERT string, simply call HiiSetString in both IF branches:

```suggestion
        AsciiStrToUnicodeStrS(CertificateInfoStr8, CertificateInfoStr, CertificateInfoStrSize/sizeof(CHAR16));
        GuidID  = HiiSetString (PrivateData->HiiHandle, 0, CertificateInfoStr, NULL);
      } else {
        StrCpyS(CertificateInfoStr, CertificateInfoStrSize/sizeof(CHAR16), UNKNOWN_CERT_PLACEHOLDER);
        GuidID  = HiiSetString (PrivateData->HiiHandle, 0, UNKNOWN_CERT_PLACEHOLDER, NULL);
      }

```

Also I would change `UNKNOWN_CERT_PLACEHOLDER` to `UNKNOWN_CERT`.